lib/AgrirouterClient/inc/ConnectionProvider.h

```diff
@@ -38,13 +38,13 @@ class ConnectionProvider
         void *getMember();
 
     protected:
-        Settings *settings;
-        std::string body;
-        std::string url;
-        std::vector<std::string> headers;
-        std::string applicationMessageId;
-        Callback callback;
-        void *member;
+        Settings *m_settings;

```

Comment:
Initialize pointer with nullptr (or NULL) to prevent undefined behaviour.
Or maybe a better solution: Declare Settings as const pointer (Settings* const m_settings) and initialize in initializer list of constructor.